### PR TITLE
update to 0.8.5

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
aws-c-common 0.8.5
Adding required cmake function
[jira](https://anaconda.atlassian.net/browse/PKG-3912)
[github](https://github.com/awslabs/aws-c-common/tree/v0.8.5)
dependency for aws-sdk-cpp 1.10.55 -> arrow-cpp 14.0.1 -> pyarrow 14.0.1  which would address CVE

### CODE REVIEW

**Comment by cbouss** [comment]
Why not targeting arrow-cpp/pyarrow 15?
In that case: https://github.com/apache/arrow/blob/apache-arrow-15.0.0/cpp/thirdparty/versions.txt#L32

**Comment by M-Waszkiewicz-Anaconda** [comment]
> Why not targeting arrow-cpp/pyarrow 15? In that case: https://github.com/apache/arrow/blob/apache-arrow-15.0.0/cpp/thirdparty/versions.txt#L32

14.0.1 is good enough already, see
https://nvd.nist.gov/vuln/detail/CVE-2023-47248

**Review by cbouss — APPROVED** [review]

**Review by lorepirri — APPROVED** [review]

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,16 +1,16 @@
...
   url: https://github.com/awslabs/{{ name }}/archive/v{{ version }}.tar.gz
-  sha256: 2997e851ed690a614507a43f4c393f45a198614d94da1660ecdf9b5a729535fc
+  sha256: 9a6b9026de9ad0c1aac2aadbbf83e331f8259d8a8d6ce2f59f2094040bba3e71
 
 build:
-  number: 1
+  number: 0
   run_exports:
     - {{ pin_subpackage("aws-c-common", max_pin="x.x.x") }}
   missing_dso_whitelist:
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:16]: It's better to use `ninja-base` (because it hasn't `python` support) instead of `ninja`

**Review by skupr-anaconda — APPROVED** [review]
LGTM but `ninja-base`

**Pull request merged**